services/provider/server/server_test.go

```diff
@@ -284,12 +285,15 @@ func TestGetExternalResources(t *testing.T) {
 	assert.NoError(t, err)
 	assert.NotNil(t, storageConRes)
 
-	for i := range mockExtR {
-		data, err := json.Marshal(mockExtR[i].Data)
+	for i := range storageConRes.ExternalResource {
+		extResource := storageConRes.ExternalResource[i]
+		mockResoruce := mockExtR[extResource.Name]
```

Comment:
Same here, take the address.
Also, you do not know if the item is in the map or not (this is what the test is checking)
So better to do:
```suggestion
		mockResoruce, ok := mockExtR[extResource.Name]
		Assert.True(ok)
```